File: JANA/JANA/lib/features/jana_ai/presentation/widgets/jana_right_panel.dart
import 'package:flutter/material.dart';
import 'package:intl/intl.dart';
import '../../application/jana_ai_view_state.dart';
import '../theme/jana_colors.dart';
import '../theme/jana_spacing.dart';
import '../theme/jana_text_styles.dart';
import 'jana_case_summary_card.dart';
import 'jana_janmitra_panel.dart';

class JanaRightPanel extends StatefulWidget {
  final JanaAiViewState state;
  final VoidCallback? onReturnToAi;

  const JanaRightPanel({Key? key, required this.state, this.onReturnToAi}) : super(key: key);

  @override
  State<JanaRightPanel> createState() => _JanaRightPanelState();
}

class _JanaRightPanelState extends State<JanaRightPanel> {
  final ScrollController _scrollController = ScrollController();

  @override
  void didUpdateWidget(JanaRightPanel oldWidget) {
    super.didUpdateWidget(oldWidget);
    if (widget.state.contextEvents.length > oldWidget.state.contextEvents.length) {
      WidgetsBinding.instance.addPostFrameCallback((_) {
        if (_scrollController.hasClients) {
          _scrollController.animateTo(
            _scrollController.position.maxScrollExtent,
            duration: const Duration(milliseconds: 500),
            curve: Curves.easeOut,
          );
        }
      });
    }
  }

  @override
  void dispose() {
    _scrollController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    // ── JANMITRA MODE: Show associate UI in the right panel ─────────────────
    if (widget.state.controlledBy == 'HUMAN') {
      return _buildJanmitraView();
    }

    // ── AI MODE: Show clinical assistant events ──────────────────────────────
    final events = widget.state.contextEvents.reversed.toList();

    return Container(
      width: double.infinity,
      decoration: const BoxDecoration(
        color: JanaColors.cardBackground,
        border: Border(left: BorderSide(color: JanaColors.divider)),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Padding(
            padding: const EdgeInsets.all(JanaSpacing.md),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                const Text('Clinical Assistant', style: JanaTextStyles.header2),
                const SizedBox(height: 4),
                Text(
                  'Real-time clinical updates',
                  style: JanaTextStyles.caption.copyWith(fontSize: 12),
                ),
              ],
            ),
          ),
          const Divider(height: 1),
          Expanded(
            child: widget.state.contextEvents.isEmpty
                ? _buildEmptyPlaceholder()
                : ListView.builder(
                    physics: const BouncingScrollPhysics(parent: AlwaysScrollableScrollPhysics()),
                    controller: _scrollController,
                    padding: const EdgeInsets.all(JanaSpacing.md),
                    itemCount: events.length,
                    itemBuilder: (context, index) {
                      final event = events[index];
                      final bool isLatest = index == 0;
                      return _buildNotificationItem(event, isLatest);
                    },
                  ),
          ),
        ],
      ),
    );
  }

  /// Janmitra Associate panel — renders inside the right panel (associate mode).
  Widget _buildJanmitraView() {
    final sessionId = widget.state.sessionId;
    final janmitra = widget.state.janmitraData;
    final name = janmitra?['fullName'] as String? ?? 'Janmitra Associate';
    final role = janmitra?['role'] as String? ?? 'Janmitra Associate';

    return Container(
      width: double.infinity,
      decoration: const BoxDecoration(
        color: Color(0xFF0E1A2B),
        border: Border(left: BorderSide(color: Color(0xFFFF9800), width: 2)),
      ),
      child: Column(
        children: [
          // Header banner
          Container(
            width: double.infinity,
            padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 12),
            decoration: const BoxDecoration(
              color: Color(0xFF1A1200),
              border: Border(bottom: BorderSide(color: Color(0xFFFF9800), width: 1)),
            ),
            child: const Row(
              children: [
                Icon(Icons.support_agent, color: Color(0xFFFF9800), size: 18),
                SizedBox(width: 8),
                Expanded(
                  child: Text(
                    'Janmitra Associate',
                    style: TextStyle(
                      color: Color(0xFFFF9800),
                      fontWeight: FontWeight.bold,
                      fontSize: 14,
                    ),
                  ),
                ),
              ],
            ),
          ),

          Expanded(
            child: sessionId != null
                ? JanaJanmitraPanel(
                    sessionId: sessionId,
                    janmitraName: name,
                    janmitraRole: role,
                    janmitraData: janmitra,
                    onReturnToAi: widget.onReturnToAi,
                  )
                : const Center(
                    child: Text('Connecting to associate...',
                        style: TextStyle(color: Colors.white38)),
                  ),
          ),
        ],
      ),
    );
  }

  Widget _buildEmptyPlaceholder() {
    return Center(
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Icon(Icons.notifications_none, color: Colors.grey[300], size: 48),
          const SizedBox(height: 16),
          Text(
            'Waiting for updates...',
            style: JanaTextStyles.caption.copyWith(color: Colors.grey[500]),
          ),
        ],
      ),
    );
  }

  Widget _buildNotificationItem(Map<String, dynamic> event, bool isLatest) {
    final String eventType = event['event_type'] ?? 'GENERIC_EVENT';
    final payload = event['payload'] ?? {};
    final String timestampStr = payload['timestamp'] ?? DateTime.now().toIso8601String();
    final DateTime time = DateTime.parse(timestampStr);
    final String formattedTime = DateFormat('hh:mm a').format(time.toLocal());

    String title = 'Update';
    String details = payload['message'] ?? 'Action processed.';
    IconData icon = Icons.notifications_active_outlined;
    Color color = JanaColors.primaryTeal;

    switch (eventType) {
      case 'MEMBER_VERIFIED':
        title = 'Patient Verified';
        icon = Icons.verified_user_outlined;
        color = Colors.blue;
        break;
      case 'CASE_CREATED':
        title = 'Case Created';
        icon = Icons.folder_shared_outlined;
        color = Colors.indigo;
        break;
      case 'APPOINTMENT_BOOKED':
        title = 'Booking Done';
        icon = Icons.event_available;
        color = Colors.deepPurple;
        break;
      case 'DOCTOR_OUTCOME_GENERATED':
        title = 'Outcome Ready';
        icon = Icons.assignment_outlined;
        color = Colors.teal;
        break;
      case 'LAB_TEST_SCHEDULED':
        title = 'Tests Booked';
        icon = Icons.biotech;
        color = Colors.cyan;
        break;
      case 'TEST_COMPLETED':
        title = 'Report Ready';
        icon = Icons.article_outlined;
        color = Colors.redAccent;
        break;
      case 'MEDICINE_DELIVERY_BOOKED':
        title = 'Dispatch Ready';
        icon = Icons.local_shipping;
        color = Colors.orange;
        break;
      case 'STEP_COMPLETED':
        title = payload['title'] ?? 'Step Complete';
        details = payload['message'] ?? 'Successfully completed.';
        icon = Icons.check_circle_outline;
        color = Colors.green;
        break;
      case 'HANDOFF':
        title = payload['title'] ?? 'Associate Joined';
        icon = Icons.support_agent;
        color = Colors.orange;
        break;
    }

    return AnimatedScale(
      scale: 1.0,
      duration: const Duration(milliseconds: 300),
      child: Container(
        margin: const EdgeInsets.only(bottom: 12),
        padding: EdgeInsets.all(isLatest ? 16 : 12),
        decoration: BoxDecoration(
          color: isLatest ? color.withOpacity(0.08) : Colors.white,
          borderRadius: BorderRadius.circular(12),
          border: Border.all(
            color: isLatest ? color.withOpacity(0.5) : JanaColors.divider,
            width: isLatest ? 2 : 1,
          ),
          boxShadow: isLatest ? [
            BoxShadow(color: color.withOpacity(0.1), blurRadius: 8, offset: const Offset(0, 4))
          ] : null,
        ),
        child: Row(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Container(
              padding: const EdgeInsets.all(8),
              decoration: BoxDecoration(
                color: isLatest ? color : color.withOpacity(0.1),
                shape: BoxShape.circle,
              ),
              child: Icon(icon, size: isLatest ? 20 : 16, color: isLatest ? Colors.white : color),
            ),
            const SizedBox(width: 12),
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Row(
                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
                    children: [
                      Text(
                        title,
                        style: TextStyle(
                          fontWeight: FontWeight.bold,
                          fontSize: isLatest ? 15 : 13,
                          color: JanaColors.darkText,
                        ),
                      ),
                      Text(
                        formattedTime,
                        style: JanaTextStyles.caption.copyWith(fontSize: 10),
                      ),
                    ],
                  ),
                  const SizedBox(height: 4),
                  Text(
                    details,
                    style: JanaTextStyles.caption.copyWith(
                      fontSize: isLatest ? 13 : 12,
                      height: 1.4,
                      color: isLatest ? Colors.black87 : Colors.grey[600],
                    ),
                  ),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }
}

File: jana-react-native/src/components/JanaRightPanel.js
import React, { useRef, useEffect } from 'react';
import { View, Text, ScrollView, StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

export default function JanaRightPanel({ contextEvents = [] }) {
  const scrollViewRef = useRef();

  useEffect(() => {
    if (scrollViewRef.current) {
      setTimeout(() => {
        scrollViewRef.current.scrollToEnd({ animated: true });
      }, 300);
    }
  }, [contextEvents]);

  const renderEvent = (event, index, isLatest) => {
    const eventType = event.event_type || 'GENERIC_EVENT';
    const payload = event.payload || {};
    const timestampStr = payload.timestamp || new Date().toISOString();
    const time = new Date(timestampStr);
    const hours = time.getHours().toString().padStart(2, '0');
    const minutes = time.getMinutes().toString().padStart(2, '0');
    const formattedTime = `${hours}:${minutes}`;

    let title = 'Update';
    let details = payload.message || 'Action processed.';
    let icon = 'notifications-outline';
    let color = '#008080';

    switch (eventType) {
      case 'MEMBER_VERIFIED':
        title = 'Patient Verified'; icon = 'checkmark-circle-outline'; color = '#2196F3'; break;
      case 'CASE_CREATED':
        title = 'Case Created'; icon = 'folder-open-outline'; color = '#3F51B5'; break;
      case 'APPOINTMENT_BOOKED':
        title = 'Booking Done'; icon = 'calendar-outline'; color = '#673AB7'; break;
      case 'DOCTOR_OUTCOME_GENERATED':
        title = 'Outcome Ready'; icon = 'document-text-outline'; color = '#009688'; break;
      case 'LAB_TEST_SCHEDULED':
        title = 'Tests Booked'; icon = 'flask-outline'; color = '#00BCD4'; break;
      case 'TEST_COMPLETED':
        title = 'Report Ready'; icon = 'document-outline'; color = '#FF5252'; break;
      case 'MEDICINE_DELIVERY_BOOKED':
        title = 'Dispatch Ready'; icon = 'bus-outline'; color = '#FF9800'; break;
      case 'STEP_COMPLETED':
        title = payload.title || 'Step Complete';
        details = payload.message || 'Successfully completed.';
        icon = 'checkmark-circle-outline'; color = '#4CAF50'; break;
      case 'HANDOFF':
        title = payload.title || 'Associate Joined';
        icon = 'headset-outline'; color = '#FF9800'; break;
    }

    return (
      <View key={event.event_id || index} style={[
        styles.eventCard, 
        isLatest && { backgroundColor: color + '15', borderColor: color + '80', borderWidth: 2 }
      ]}>
        <View style={[styles.iconCircle, { backgroundColor: isLatest ? color : color + '20' }]}>
          <Ionicons name={icon} size={isLatest ? 20 : 16} color={isLatest ? '#fff' : color} />
        </View>
        <View style={styles.eventTextContainer}>
          <View style={styles.eventHeader}>
            <Text style={[styles.eventTitle, isLatest && { fontSize: 15 }]}>{title}</Text>
            <Text style={styles.eventTime}>{formattedTime}</Text>
          </View>
          <Text style={[styles.eventDetails, isLatest && { fontSize: 13, color: '#333' }]}>{details}</Text>
        </View>
      </View>
    );
  };

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.headerTitle}>Clinical Assistant</Text>
        <Text style={styles.headerSub}>Real-time clinical updates</Text>
      </View>
      <View style={styles.divider} />
      
      {contextEvents.length === 0 ? (
        <View style={styles.emptyState}>
          <Ionicons name="notifications-outline" size={48} color="#e0e0e0" />
          <Text style={styles.emptyText}>Waiting for updates...</Text>
        </View>
      ) : (
        <ScrollView 
          ref={scrollViewRef}
          style={styles.list} 
          contentContainerStyle={styles.listContent}
        >
          {contextEvents.map((ev, i) => renderEvent(ev, i, i === contextEvents.length - 1))}
        </ScrollView>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    borderLeftWidth: 1,
    borderLeftColor: 'rgba(0,0,0,0.05)',
  },
  header: {
    padding: 16,
  },
  headerTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    color: '#2c3e50',
  },
  headerSub: {
    fontSize: 12,
    color: '#7f8c8d',
    marginTop: 4,
  },
  divider: {
    height: 1,
    backgroundColor: 'rgba(0,0,0,0.05)',
  },
  list: {
    flex: 1,
  },
  listContent: {
    padding: 16,
  },
  emptyState: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  emptyText: {
    marginTop: 16,
    color: '#95a5a6',
    fontSize: 14,
  },
  eventCard: {
    flexDirection: 'row',
    padding: 12,
    backgroundColor: '#fff',
    borderRadius: 12,
    borderWidth: 1,
    borderColor: 'rgba(0,0,0,0.05)',
    marginBottom: 12,
  },
  iconCircle: {
    width: 36,
    height: 36,
    borderRadius: 18,
    alignItems: 'center',
    justifyContent: 'center',
    marginRight: 12,
  },
  eventTextContainer: {
    flex: 1,
  },
  eventHeader: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 4,
  },
  eventTitle: {
    fontWeight: 'bold',
    fontSize: 13,
    color: '#2c3e50',
  },
  eventTime: {
    fontSize: 10,
    color: '#95a5a6',
  },
  eventDetails: {
    fontSize: 12,
    color: '#7f8c8d',
    lineHeight: 18,
  }
});
